api/src/main/java/io/smallrye/reactive/messaging/annotations/EmitterFactoryFor.java

```diff
@@ -29,4 +31,27 @@
      * @return the emitter interface
      */
     Class<?> value();
+
+    final class EmitterFactoryForLiteral extends AnnotationLiteral<EmitterFactoryFor> implements EmitterFactoryFor {
+
+        private final Class<?> value;
+
+        public static EmitterFactoryFor of(Class<?> value) {
+            return new EmitterFactoryForLiteral(value);
+        }
+
+        private EmitterFactoryForLiteral(Class<?> value) {
+            this.value = value;
+        }
+
+        @Override
+        public Class<?> value() {
+            return value;
+        }
+
+        public static EmitterFactoryFor EMITTER = of(org.eclipse.microprofile.reactive.messaging.Emitter.class);
+        public static EmitterFactoryFor MUTINY_EMITTER = of(MutinyEmitter.class);
+        public static EmitterFactoryFor LEGACY_EMITTER = of(io.smallrye.reactive.messaging.annotations.Emitter.class);
```

Comment:
Nothing wrong with these literals being here, but do we want to make them part of the public API?

Reply:
If the Literal becomes the public API, I don't see why these wouldn't.